src/histolab/tiler.py

```diff
@@ -158,16 +161,35 @@ def _tile_filename(
 
         return tile_filename
 
-    def _tiles_generator(self, slide: Slide) -> Tuple[Tile, CoordinatePair]:
+    def _tiles_generator(self, slide: Slide) -> None:
         raise NotImplementedError
 
-    @lazyproperty
-    def _tiles_locator(self):
-        return (
-            self._highest_score_tiles
-            if isinstance(self, ScoreTiler)
-            else self._tiles_generator
-        )
+    def _validate_levels(self, slide: Slide) -> None:
```

Comment:
I would prefer to have a method that returns a bool whether the level is valid or not and then let the caller decide what to do with it (it may not want to raise an exception), as we did with `_has_valid_tile_size`. Moreover the method name is not coherent with what it does because we are also checking the tile size here, but with the previous suggestion it should fix itself automatically

Reply:
having a boolean check doesn't reduce the code fragment duplication. Considering that we do the same check into the 3 classes i'd like to have a sort of validator that raises those exception when we call the `extract`

Reply:
mm ok, since it is internal ok.. anyway we can always refactor it if needed.

anyway `_validate_levels` validates the tile size as well 🤷‍♀️ so I would change the name

Reply:
ok ,,, any suggestion on the name?

Reply:
`_validate_level_tile_size` ?
... we have a method that does two things

Reply:
ok maybe we can split it in 2 validators

what do you think?

Reply:
yes better